src/state_machine.zig

```diff
@@ -304,7 +304,8 @@ pub fn StateMachineType(
 
         // Since prefetch contexts are used one at a time, it's safe to access
         // the union's fields and reuse the same memory for all context instances.
-        const PrefetchContext = extern union {
```

Comment:
Hm, why did we use `extern union` in the first place here?

Reply:
To treat it as a C union to save space iirc.